DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -213,8 +153,8 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
 
             existingDevice.DeviceProperties.HubEnabledState = isEnabled;
             existingDevice.DeviceProperties.UpdatedTime = DateTime.UtcNow;
-
-            return (await _docDbRestUtil.UpdateDocumentAsync(existingDevice)).ToObject<DeviceModel>();
+            await _documentClient.SaveAsync(deviceId, existingDevice);
```

Comment:
device.id is document id and not device id. We should be passing device.DeviceProperties.DeviceID everywhere